scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.examples;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+import org.apache.commons.net.ftp.FTPReply;
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+
+
+/**
+ * Import and export file(s) using FTPConnector server.
+ * This task can be launched with parameters.
+ *
+ * @author The ProActive Team
+ */
+public class FTPConnector extends JavaExecutable {
+
+    private static final String DEFAULT_FTP_HOSTNAME = "localhost";
+
+    private static final int DEFAULT_FTP_PORT = 21;
+
+    private String ftpHostname = DEFAULT_FTP_HOSTNAME;
+
+    private int ftpPort = DEFAULT_FTP_PORT;
+
+    private boolean ftpExtractArchive;
+
+    private String ftpLocalFilePath = null;
+
+    private String ftpRemoteFilePath = null;
+
+    private String ftpMode = null;
+
+    private String ftpUsername = null;
+
+    private String ftpPassword = null;
+
+    private static final String FTP_LOCAL_FILE_PATH = "ftpLocalFilePath";
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("ftpHostname")) {
+            ftpHostname = args.get("ftpHostname").toString();
+        }
+        if (args.containsKey("ftpPort")) {
+            try {
+                ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+            } catch (NumberFormatException e) {
+            }
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey(FTP_LOCAL_FILE_PATH)) {
+            ftpLocalFilePath = args.get(FTP_LOCAL_FILE_PATH) == null ? "" : args.get(FTP_LOCAL_FILE_PATH).toString();
```

Comment:
you can put private String  ftpLocalFilePath = ""

Reply:
instead of using "" as the default value, it would be better to use the value returned by getLocalSpace()

Reply:
But it is empty string

Reply:
using empty string as default will only allow to copy files to the current directory. This will work in fork mode, but not in non-fork mode where the current directory is not writeable (and should not be written to).

That's why I suggest to use a default value based on what is returned by a call to getLocalSpace(). 

The localspace will always be writable and moreover can be used to transfer files to another data space (global , user).

Reply:
done